src/cascadia/TerminalSettingsModel/ActionMap.cpp

```diff
@@ -18,12 +19,31 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
         size_t hashedArgs{};
         if (const auto& args{ actionAndArgs.Args() })
         {
+            // Args are defined, so hash them
             hashedArgs = gsl::narrow_cast<size_t>(args.Hash());
         }
         else
         {
-            std::hash<IActionArgs> argsHash;
-            hashedArgs = argsHash(nullptr);
+            // Args are not defined.
+            // Check if the ShortcutAction supports args.
+            switch (actionAndArgs.Action())
+            {
+#define ON_ALL_ACTIONS_WITH_ARGS(action)                                    \
+    case ShortcutAction::action:                                            \
+    {                                                                       \
+        /* If it does, hash the default values for the args.*/              \
+        hashedArgs = gsl::narrow_cast<size_t>(make<action##Args>().Hash()); \
+        break;                                                              \
+    }

```

Comment:
You only need those curly braces if you introduce local variables. The default case does, but these regular cases don't.